app.py

```diff
@@ -93,10 +99,11 @@ def get_user(id):
 
 @app.route('/users/<id>', methods=['DELETE'])
 def delete_user(id):
-    user = User.query.get_or_404(id)
-    user.deleted = True
-    db.session.commit()
-    return '', 204
+    if User.query.filter(id=id).soft_delete():
```

Comment:
Okay, so to delete an object for real I have to use:

    user = User.query.get(id)
    db.session.delete(user)

But to soft delete the user I have to do something completely different:

    User.query.filter_by(id=id).soft_delete()

This doesn't really look that good, don't you think? You'd think the way to soft delete the object would be:

    user = User.query.get(id)
    db.session.soft_delete(user)

Right?

Also, as a side note, even after so many mistakes you are still not testing the code that you are submitting in this PR. The `filter(id=id)` construct is incorrect, I believe, should be `filter_by(id=id)` or `filter(User.id == id)`.

Reply:
> Okay, so to delete an object for real I have to use:

```
user = User.query.get(id)
db.session.delete(user)
```

I mean that's one way you can do it, but you can also do 
`User.query.filter_by(id=id).delete()`

which ends up being very similar to 
`User.query.filter_by(id=id).soft_delete()`


I suppose that a more complete solution would include adding a soft_delete function to the session object as well. 

Ultimately the improvement that I wanted to bring to your initial solution (which was to set the `deleted` attribute of the model to `true`) was to be able to apply it to more than one model. 

> Also, as a side note, even after so many mistakes you are still not testing the code that you are submitting in this PR. The filter(id=id) construct is incorrect, I believe, should be filter_by(id=id) or filter(User.id == id)

You are correct. I'm making my changes in the github editor, and testing the relevant code (the soft_delete) in a different application. I always manage to confuse `filter` and `filter_by`.